weldx/core.py

```diff
@@ -365,10 +364,20 @@ def _check_data_array(data_array: xr.DataArray):
         if not isinstance(data_array.data, pint.Quantity):
             raise TypeError("The data of the 'DataArray' must be a 'pint.Quantity'.")
 
+    @staticmethod
+    def _create_data_array(
+        data: Union[pint.Quantity, xr.DataArray], time: Time
+    ) -> xr.DataArray:
+        return (
+            xr.DataArray(data=data)
+            .rename({"dim_0": "time"})
+            .assign_coords({"time": time.as_timedelta_index()})
```

Comment:
In cases like these I would really like if we could attach the reference time to the xarray coordinates still

```suggestion
            xr.DataArray(data=data)
            .rename({"dim_0": "time"})
            .assign_coords({"time": time.as_data_array().weldx.time_ref_restore()})
```
(thinking about it the `.weldx.time_ref_restore()` call should probably be part of `as_data_array()`)

For the `TimeSeries` `reference_time` could then also point to the stored attribute instead of `self._reference_time`

Reply:
Can we do this in a separate PR? I already have a list of todos for the `TimeSeries` because of its new support for absolute times. I will open a new issue related to this soon and we can add it as a to-do there.

Reply:
sure

Reply:
see https://github.com/BAMWelDX/weldx/issues/471